lib/p2p/PeerList.ts

```diff
@@ -13,17 +13,22 @@ class PeerList {
     }
   }
 
-  public remove(peer:Peer): void {
+  public remove = (peer:Peer): void => {
     delete this.peers[peer.socketAddress.toString()];
   }
 
-  public has(socketAddress: SocketAddress): boolean {
+  public has = (socketAddress: SocketAddress): boolean => {
     return !!this.peers[socketAddress.toString()];
   }
 
-  public forEach(cb: (peer: Peer) => void) {
+  public length = () => {
```

Comment:
Might be nice to make this a getter function. That way we can use `peers.length` without the () which I think is more consistent.

EDIT: Actually, better yet see below about moving this logic to the `Pool` class.